Hydro-Grapher_AWDR_GWSI_v1.12_TRANSDUCERS_20191216.py

```diff
@@ -120,45 +126,22 @@
     for tick in ax1.get_xticklabels():
         tick.set_rotation(90)
 
-    #SET X-AXIS LIMITS (xlim)
-    ax1.set_xlim([min_date_4fig,max_date_4fig])
+    # SET X-AXIS LIMITS (xlim)
+    ax1.set_xlim([min_date_4fig, max_date_4fig])
 
-    ax1.xaxis.set_major_locator(dates.YearLocator(1))#THIS WORKS
+    ax1.xaxis.set_major_locator(dates.YearLocator(1))  # THIS WORKS
 
     plt.rcParams.update({'font.size': 12})
 
-    plt.show()
+    if SHOW:
+        plt.show() # TODO remove this, it significantly slows down the program.
 
     outname = str('Hydrographs_GWSI_Automated__') + str(location) + str('.png')
-    fig.savefig(outname, dpi = 400, bbox_inches='tight', pad_inches=.1)
-
-#    outname = str('GWSI_WLE_ZipExtract_Seasonal_') + str(location) + str('__Raw_Data_Table.csv')
-#    df4.to_csv(outname, index=False)
-#
-#    outname = str('_Error_Date_Bad.csv')
-#    df.to_csv(outname, index=False)
-##############################################################################
-##############################################################################
-##############################################################################
-#  ### ### ### ### ### #### ### ### ### ### ### #### ### ### ### ### ### ###  #
-##   Example and Test Code Used for This Program   ##
-"""
-outname = str('test_df4.csv')
-df4.to_csv(outname, index=False)
+    fig.savefig(outname, dpi=400, bbox_inches='tight', pad_inches=.1)
 
 
-###seasonal trending wells with high variation:
-#wells = [315614110543601, 315705110560901, 315909110540601, 320216110405901, 320249110413501, 320944111125701, 321034111140801, 321058110563301, 321244110504401, 321348111154401, 321404111153201, 321414111133601, 321424110432901, 321445110554601, 321456111160201, 321503110484901, 321504110521601, 321512110480701, 321523110491801, 321530110523501, 321537110585001, 321542110502701, 321548110561801, 321601110544601, 321703110575401, 321706110442201, 321745111012801, 321748110582101, 322238111065701, 322339111170001, 323635111330001]
+with Pool(cpu_count() / 2) as pool:

```

Comment:
If you really want to speed it up we could use dask and a distributed number of systems.